pkg/resource/resource.go

```diff
@@ -116,3 +123,42 @@ func (r *BaseResource) ID() string {
 
 	return r.id
 }
+
+func (r *BaseResource) Logs() []logger.Entry {
+	return r.logs
+}
+
+func (r *BaseResource) Fatal(code logger.ErrorCode, msg string, args ...any) {
+	entry := logger.Error(msg, args...)
+	entry.Code = code
+	entry.Fatal = true
+	r.logs = append(r.logs, entry)
+}
+
+func (r *BaseResource) Debug(code logger.ErrorCode, msg string, args ...any) {
+	entry := logger.Debug(msg, args...)
+	entry.Code = code
+	entry.Fatal = false
+	r.logs = append(r.logs, entry)
+}
+
+func (r *BaseResource) Error(code logger.ErrorCode, msg string, args ...any) {
+	entry := logger.Error(msg, args...)
+	entry.Code = code
+	entry.Fatal = false
+	r.logs = append(r.logs, entry)
```

Comment:
If you had logger.Error only create the entry if the logging level is set to a certain level like before and it return a pointer to the entry or nil, this could be come:

```suggestion
	if entry := logger.Error(msg, args...); entry != nil {
		entry.Code = code
		r.logs = append(r.logs, entry)
	}
```

and then you'd only get logs on the resource if the log level allows for it.

Reply:
I originally had it this way but then when we have a different log level vs resource log level we would come unstuck. It could be that the resource gets all the logs but the stderr does not and vice versa.

Reply:
Gotcha. Yeah I guess the idea of avoiding making a bunch of extra structs was what made me think about doing it that way.

Thoughts on resource log level == normal log level for now and there's no real separating the two? And if we need to do per-scan logging levels, we tweak it?

Reply:
We can do that for now. 
Then we will just have a config for whether to have non critical logs included in responses.